File: rb/features.rb
# TempMail SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module TempMailFeatures
  def self.make_feature(name)
    case name
    when "base"
      TempMailBaseFeature.new
    when "ratelimit"
      TempMailRatelimitFeature.new
    when "retry"
      TempMailRetryFeature.new
    when "test"
      TempMailTestFeature.new
    when "timeout"
      TempMailTimeoutFeature.new
    else
      TempMailBaseFeature.new
    end
  end
end

File: py/features.py
# TempMail SDK feature factory

from feature.base_feature import TempMailBaseFeature
from feature.test_feature import TempMailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TempMailBaseFeature(),
        "test": lambda: TempMailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
